source/common/router/rds_impl.cc

```diff
@@ -219,6 +261,16 @@ void RdsRouteConfigProviderImpl::validateConfig(
   ConfigImpl validation_config(config, factory_context_, validator_, false);
 }
 
+// Schedules a VHDS request on the main thread and queues up the callback to use when the VHDS
+// response has been propagated to the worker thread that was the request origin.
+void RdsRouteConfigProviderImpl::requestVirtualHostsUpdate(
+    const std::string& for_domain, std::function<void()> route_config_updated_cb) {
+  factory_context_.dispatcher().post(
```

Comment:
The filter chain for which the instance of a RdsRouteConfigProvider has been created is paused after ```requestVirtualHostsUpdate``` call is made (please see https://github.com/envoyproxy/envoy/pull/8617/files#diff-53e919b583d3290e6fddf324e7e69885R23). It will be resumed only after a response to this request arrives and a callback which calls https://github.com/envoyproxy/envoy/pull/8617/files#diff-53e919b583d3290e6fddf324e7e69885R44 is triggered.

Please let me know if this level of details is sufficient or you'd like more information.